### PR TITLE
Migrate from truffle to hardhat - Close #AM-2912

### PR DESCRIPTION
**Pull request by zarifpour**
- Remove truffle-config.js
- Remove migrations/1_deploy_contracts.js
- Add scripts/1_deploy_contracts.js
- Update hardhat.config.js with network configurations 
- Update ESLint configuration, environment, and globals Update devDependencies in package.json
- Update deployment.test.js with new deployment logic Update .env.example with private key and node URLs

Close #AM-2912

> [!Note]
> We should merge #17 first, which will add slither to our CI and add the SARIF to code analysis.

### CODE REVIEW

**Comment by zarifpour** [comment]
Sample of deployments (x2): https://goerli.etherscan.io/address/0x1e874187bef4b3bc59a906b5838a6f3ea849bb6f

<details>
  <summary>console logs</summary>

```
❯ npx hardhat run scripts/1_deploy_contracts.js --network goerli

+=================================+
|  Deploying CMTAT_BASE Contract  |
+=================================+

Deployed contract object:
 BaseContract {
  target: '0xb49f2585541cCaEc347F1B1Aa318027fCfEbdffc',
  interface: Interface {
    fragments: [
      [ConstructorFragment], [ErrorFragment],    [ErrorFragment],
      [ErrorFragment],       [EventFragment],    [EventFragment],
      [EventFragment],       [EventFragment],    [EventFragment],
      [EventFragment],       [EventFragment],    [EventFragment],
      [EventFragment],       [EventFragment],    [EventFragment],
      [EventFragment],       [EventFragment],    [EventFragment],
      [EventFragment],       [FunctionFragment], [FunctionFragment],
      [FunctionFragment],    [FunctionFragment], [FunctionFragment],
      [FunctionFragment],    [FunctionFragment], [FunctionFragment],
      [FunctionFragment],    [FunctionFragment], [FunctionFragment],
      [FunctionFragment],    [FunctionFragment], [FunctionFragment],
      [FunctionFragment],    [FunctionFragment], [FunctionFragment],
      [FunctionFragment],    [FunctionFragment], [FunctionFragment],
      [FunctionFragment],    [FunctionFragment], [FunctionFragment],
      [FunctionFragment],    [FunctionFragment], [FunctionFragment],
      [FunctionFragment],    [FunctionFragment], [FunctionFragment],
      [FunctionFragment],    [FunctionFragment], [FunctionFragment],
      [FunctionFragment],    [FunctionFragment], [FunctionFragment]
    ],
    deploy: ConstructorFragment {
      type: 'constructor',
      inputs: [],
      payable: false,
      gas: null
    },
    fallback: null,
    receive: false
  },
  runner: HardhatEthersSigner {
    _gasLimit: undefined,
    address: '0x1E874187BeF4b3BC59A906B5838A6f3Ea849bB6f',
    provider: HardhatEthersProvider {
      _hardhatProvider: [LazyInitializationProviderAdapter],
      _networkName: 'goerli',
      _blockListeners: [],
      _transactionHashListeners: Map(0) {},
      _eventListeners: []
    }
  },
  filters: {},
  fallback: null,
  deploymentTransaction: [Function (anonymous)],
  [Symbol(_ethersInternal_contract)]: {}
}
Proxy deployed at: 0xb49f2585541cCaEc347F1B1Aa318027fCfEbdffc

Waiting 30 seconds for the network to propagate...

+============================+
|  Verifying Proxy Contract  |
+============================+

Proxy contract address: 0xb49f2585541cCaEc347F1B1Aa318027fCfEbdffc

Verifying contract on explorer...
The contract 0xb49f2585541cCaEc347F1B1Aa318027fCfEbdffc has already been verified
Verification submitted for contract at address: 0xb49f2585541cCaEc347F1B1Aa318027fCfEbdffc

+==================================+
|  Verifying Proxy Admin Contract  |
+==================================+

Proxy Admin contract address: 0x8fD118Fd02A27a31BA824bCA304F26060Bc4bDC8

Verifying contract on explorer...
The contract 0x8fD118Fd02A27a31BA824bCA304F26060Bc4bDC8 has already been verified
Verification submitted for contract at address: 0x8fD118Fd02A27a31BA824bCA304F26060Bc4bDC8

+=====================================+
|  Verifying Implementation Contract  |
+=====================================+

Implementation contract address: 0x3F3161F82894C71874D26d3322AfD3394279eb04

Verifying contract on explorer...
The contract 0x3F3161F82894C71874D26d3322AfD3394279eb04 has already been verified
Verification submitted for contract at address: 0x3F3161F82894C71874D26d3322AfD3394279eb04

+======================+
|  Deployment Summary  |
+======================+

Deployed Proxy Contract Address: 0xb49f2585541cCaEc347F1B1Aa318027fCfEbdffc
Proxy Admin Contract Address: 0x8fD118Fd02A27a31BA824bCA304F26060Bc4bDC8
Implementation Contract Address: 0x3F3161F82894C71874D26d3322AfD3394279eb04
```
</details>

**Comment by zarifpour** [comment]
I suggest creating a new branch and merging to it, i.e. `wrapped-assest-hh`, to preserve our truffle codebase just in case. We can experiment there and decide if we would like to do a complete migration.

File: scripts/1_deploy_contracts.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    } catch (error) {
+      console.error(`Error verifying ${contractType} contract:`, error)
+    }
+  }
+
+  // Verify Proxy Contract
+  await performVerification(() => verifyProxyContract(proxyContract, constructorArguments), 'Proxy')
+
+  // Verify Proxy Admin Contract
+  await performVerification(() => verifyProxyAdminContract(proxyContract), 'Proxy Admin')
```
Comment (diego-G) [inline scripts/1_deploy_contracts.js:63]: Add an `if` statement when the ProxyAdmin and implementation were not deployed and therefore are already verified.

**Review by diego-G — CHANGES_REQUESTED** [review]
Minimal changes

File: scripts/1_deploy_contracts.js (hunk 1 above)
Comment (zarifpour) [reply scripts/1_deploy_contracts.js:63]: this should be okay because it is a try catch

File: hardhat.config.js [hunk 2]
```diff
@@ -1,8 +1,49 @@
-/** @type import('hardhat/config').HardhatUserConfig */
-require('@nomiclabs/hardhat-truffle5')
+require('dotenv').config()
+require('hardhat-gas-reporter')
 require('solidity-coverage')
 require('solidity-docgen')
+require('@openzeppelin/hardhat-upgrades')
+require('@nomiclabs/hardhat-etherscan')
```
Comment (zarifpour) [inline hardhat.config.js]: this was deprecated... need to move to hardhat-verify

**Review by diego-G — DISMISSED** [review]

**Review by mariogutval — DISMISSED** [review]

**Comment by diego-G** [comment]
I've just realised that we still have truffle in several package scripts, `usage.md` and `toolchain.md` files. Please @zarifpour, replace it by Hardhat.

**Comment by zarifpour** [comment]
Please review the last commit

**Review by diego-G — APPROVED** [review]

**Pull request merged**